src/Microsoft.Health.Dicom.Core/Features/Store/Entries/DicomInstanceEntryReaderForSinglePartRequest.cs

```diff
@@ -22,12 +28,16 @@ namespace Microsoft.Health.Dicom.Core.Features.Store.Entries
     public class DicomInstanceEntryReaderForSinglePartRequest : IDicomInstanceEntryReader
     {
         private readonly ISeekableStreamConverter _seekableStreamConverter;
+        private readonly IOptions<StoreConfiguration> _storeConfiguration;
 
-        public DicomInstanceEntryReaderForSinglePartRequest(ISeekableStreamConverter seekableStreamConverter)
+
+        public DicomInstanceEntryReaderForSinglePartRequest(ISeekableStreamConverter seekableStreamConverter, IOptions<StoreConfiguration> storeConfiguration)
         {
             EnsureArg.IsNotNull(seekableStreamConverter, nameof(seekableStreamConverter));
+            EnsureArg.IsNotNull(storeConfiguration, nameof(storeConfiguration));
 
             _seekableStreamConverter = seekableStreamConverter;
+            _storeConfiguration = storeConfiguration;
```

Comment:
Store the configuration and not the options object. Calling the `Value` property resolve the underlying configuration binding and we can eagerly perform that in the ctor